nad_ch/infrastructure/auth.py

```diff
@@ -18,26 +24,59 @@ def fetch_oauth2_token(self, provider_name: str, code: str) -> str | None:
         if not provider_config:
             return None
 
-        token_url = provider_config["token_url"]
-        request_data = {
-            "client_id": provider_config["client_id"],
-            "client_secret": provider_config["client_secret"],
-            "code": code,
-            "grant_type": "authorization_code",
-            "redirect_uri": url_for(
-                "auth.oauth2_callback",
-                provider=provider_name,
-                _scheme=self._callback_url_scheme,
-                _external=True,
-            ),
-        }
+        if provider_name == "logingov":
```

Comment:
I wonder if this should be branched off of the strategy login.gov uses, rather than this config identifier. i.e., check if it's a JWT oauth strategy. Alternately, maybe just delete the old cloud.gov logic? We shouldn't need it...

Reply:
Sure, I like the idea of getting rid of `cloud.gov` authentication logic altogether.